server/audit.py

```diff
@@ -0,0 +1,119 @@
+import datetime
+from flask import json, request
+from flask_admin.contrib.sqla import ModelView
+from flask_login import current_user
+from markupsafe import Markup
+from sqlalchemy import DateTime, event
+from sqlalchemy.orm.attributes import get_history
+from .database import db
+import enum
+
+
+class AuditAction(enum.Enum):
+    INSERT = "insert"
+    UPDATE = "update"
+    DELETE = "delete"
+
+
+class AuditLog(db.Model):
+    """Audit log to track data history."""
+
+    __tablename__ = "audit_log"
+
+    id = db.Column(db.Integer(), primary_key=True)
+    model_name = db.Column(db.String(), nullable=False)
```

Comment:
An enum of audited models/tables would use less DB storage. The audit log is going to be one of the biggest tables, so this is a place where shaving bytes might actually pay off. An enum could also be used as a _filter_ for which models/tables actually get audit logs. There may be some tables that you _don't_ want audit logs for.